sensor/kubernetes/listener/resources/secrets_test.go

```diff
@@ -369,3 +372,47 @@ func getImageIntegrationEvents(events *component.ResourceEvent) []*central.Senso
 
 	return iiEvents
 }
+
+// TestSkipIntegrationCreate verifies that integrations are NOT created in
+// various condition.
+func TestSkipIntegrationCreate(t *testing.T) {
+	globalPullSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      openshift.GlobalPullSecretName,
+			Namespace: openshift.GlobalPullSecretNamespace,
+		},
+	}
+	globalPullSecretAlt := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "installation-pull-secrets",
+			Namespace: "openshift-image-registry",
+		},
+	}
+
+	// No integration creations should be skipped for these test cases.
+	noSkipTcs := []struct {
+		sourcedAutogenEnabled bool
+		globalPullEnabled     bool
+	}{
+		{},
+	}
+	for i, tc := range noSkipTcs {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			testutils.MustUpdateFeature(t, features.SourcedAutogeneratedIntegrations, tc.sourcedAutogenEnabled)
+			t.Setenv(env.AutogenerateGlobalPullSecRegistries.EnvVar(), strconv.FormatBool(tc.globalPullEnabled))
+
+			assert.False(t, skipIntegrationCreate(globalPullSecret))
+			assert.False(t, skipIntegrationCreate(globalPullSecretAlt))
+		})
+	}
```

Comment:
I feel like we should add a few test cases here:

1. If `SourcedAutogeneratedIntegrations` is enabled, then we will never skip.
2. ~If `SourcedAutogeneratedIntegrations` is disabled and`AutogenerateGlobalPullSecRegistries` is enabled, then `globalPullSecret` will not be skipped but `globalPullSecretAlt` will.~ Edit: I saw you have this case below.

Reply:
Oh wow, I missed filling those test cases in entirely - nice catch.